ColorDeductoActivity.py

```diff
@@ -73,8 +73,19 @@ def __init__(self, handle):
 
         self._sharing = False
         self._initiating = False
-        self.connect('shared', self._shared_cb)
-        self.connect('joined', self._joined_cb)
+
+        def on_activity_joined_cb(me):
+            logging.debug('activity joined')
+
+        self.connect('joined', on_activity_joined_cb)
+
+        def on_activity_shared_cb(me):
+            logging.debug('activity shared')
+            self.status.set_label(_('Wait for other players to join.'))
+            self.after_share_join(True)
+
+        self.connect('shared', on_activity_shared_cb)
```

Comment:
This is the same as before but only this time they're different methods, the idea was if you're using collabwrapper for collaboration, keep those signal connections to collabwrapper and not use both sugar and collabwrapper for the calls.

Reply:
@chimosky I am bit confused. Can we not just use collabwrapper signals ? As CollabWrapper is used for collaboration, we should just call the joined event of collabwrapper and not the one from the sugar3.activity.activity.Activity class.

Reply:
Yes as you're using collabwrapper, the signal calls should be from collabwrapper and not from `sugar3.activity.activity.Activity`.